lib/archive/unpack.go

```diff
@@ -73,64 +67,54 @@ func Unpack(op trace.Operation, tarStream io.Reader, filter *FilterSpec, unpackP
 			// This indicates the end of the archive
 			break
 		}
+
 		if err != nil {
-			// it is likely in this case that we were not given a legitimate tar stream
-			op.Debugf("Received error (%s) when attempting a tar read operation on target stream", err)
+			op.Errorf("Error reading tar header: %s", err)
 			return err
 		}
 
+		op.Debugf("processing tar header: %s", header.Name)
+
 		// skip excluded elements unless explicitly included
-		if excluded(header.Name, filter) {
+		if filter.Excludes(op, header.Name) {
 			continue
 		}
 
 		// fix up path
-		strippedTargetPath := strings.TrimPrefix(header.Name, strip)
-		writePath := filepath.Join(finalTargetPath, strippedTargetPath)
+		stripped := strings.TrimPrefix(header.Name, filter.StripPath)
```

Comment:
Don't assume there are todo's everywhere where filterspec updates are needed - that needs vetting and testing in the filterspec PR.